src/catalog/table_schema.cpp

```diff
@@ -0,0 +1,198 @@
+    }
+}
+
+void Property::serialize(FileInfo* fileInfo, uint64_t& offset) const {
+    SerDeser::serializeValue<std::string>(name, fileInfo, offset);
+    SerDeser::serializeValue<LogicalType>(dataType, fileInfo, offset);
+    SerDeser::serializeValue<property_id_t>(propertyID, fileInfo, offset);
+    SerDeser::serializeValue<table_id_t>(tableID, fileInfo, offset);
+}
+
+std::unique_ptr<Property> Property::deserialize(FileInfo* fileInfo, uint64_t& offset) {
+    std::string name;
+    LogicalType dataType;
+    property_id_t propertyID;
+    table_id_t tableID;
+    SerDeser::deserializeValue<std::string>(name, fileInfo, offset);
+    SerDeser::deserializeValue<LogicalType>(dataType, fileInfo, offset);
+    SerDeser::deserializeValue<property_id_t>(propertyID, fileInfo, offset);
+    SerDeser::deserializeValue<table_id_t>(tableID, fileInfo, offset);
+    return std::make_unique<Property>(name, dataType, propertyID, tableID);
+}
+
+bool TableSchema::isReservedPropertyName(const std::string& propertyName) {
+    return StringUtils::getUpper(propertyName) == InternalKeyword::ID;
+}
+
+std::string TableSchema::getPropertyName(property_id_t propertyID) const {
+    for (auto& property : properties) {
+        if (property.propertyID == propertyID) {
+            return property.name;
+        }
+    }
+    throw RuntimeException(StringUtils::string_format(
+        "Table: {} doesn't have a property with propertyID={}.", tableName, propertyID));
+}
+
+property_id_t TableSchema::getPropertyID(const std::string& propertyName) const {
+    for (auto& property : properties) {
+        if (property.name == propertyName) {
+            return property.propertyID;
+        }
+    }
+    throw RuntimeException(StringUtils::string_format(
+        "Table: {} doesn't have a property with propertyName={}.", tableName, propertyName));
+}
+
+Property TableSchema::getProperty(property_id_t propertyID) const {
+    for (auto& property : properties) {
+        if (property.propertyID == propertyID) {
+            return property;
+        }
+    }
+    throw RuntimeException(StringUtils::string_format(
+        "Table: {} doesn't have a property with propertyID={}.", tableName, propertyID));
+}
+
+void TableSchema::renameProperty(property_id_t propertyID, const std::string& newName) {
+    for (auto& property : properties) {
+        if (property.propertyID == propertyID) {
+            property.name = newName;
+            return;
+        }
+    }
+    throw InternalException(
+        StringUtils::string_format("Property with id={} not found.", propertyID));
+}
+
+void TableSchema::serialize(FileInfo* fileInfo, uint64_t& offset) {
+    SerDeser::serializeValue<std::string>(tableName, fileInfo, offset);
+    SerDeser::serializeValue<table_id_t>(tableID, fileInfo, offset);
+    SerDeser::serializeValue<TableType>(tableType, fileInfo, offset);
+    SerDeser::serializeValue<uint64_t>(properties.size(), fileInfo, offset);
+    for (auto& property : properties) {
```

Comment:
You can reuse the serializeVector interface in ser/deser

Reply:
Added `serializeVectorOfObjects` and `deserializeVectorOfObjects` in ser/deser for now.